app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb

```diff
@@ -220,10 +196,109 @@ def self.params_for_create(ems)
           :onText     => _('Yes'),
           :offText    => _('No'),
           :isRequired => true,
+        }
+      ]
+    }
+  end
+
+  def params_for_update
+    {
+      :fields => [
+        {
+          :component  => 'text-field',
+          :name       => 'size',
+          :id         => 'size',
+          :label      => _('Size (in bytes)'),
+          :type       => 'number',
+          :step       => 1.gigabytes,
+          :isDisabled => !!id,
+          :isRequired => true,
+          :validate   => [{:type => 'required'}, {:type => 'min-number-value', :value => 0, :message => _('Size must be greater than or equal to 0')}],
           :condition  => {
-            :when => 'edit',
-            :is   => false,
+            :not => {
+              :when => 'volume_type',
+              :is   => 'standard',
+            },
           },
+        },
+        {
+          :component  => 'select',
+          :name       => 'availability_zone_id',
+          :id         => 'availability_zone_id',
+          :label      => _('Availability Zone'),
+          :isDisabled => !!id,
+          :options    => ext_management_system.availability_zones.map do |az|
+            {
+              :label => az.name,
+              :value => az.id,
+            }
+          end,
+          :isRequired => true,
+          :validate   => [{:type => 'required'}],
+        },
+        {
+          :component  => 'select',
+          :name       => 'volume_type',
+          :id         => 'volume_type',
+          :label      => _('Cloud Volume Type'),
+          :isDisabled => !!id,
+          :options    => CLOUD_VOLUME_TYPES.map do |value, label|
```

Comment:
Same comment here, why loop through all of the options when one is already selected and cannot be changed?

Reply:
I fixed the base snapshot so that it won't have to load all the options but wasn't sure how to fix this one.